client/src/components/License/LicenseSelector.vue

```diff
@@ -1,130 +1,161 @@
+<script setup lang="ts">
+import { faEdit, faSave, faTimes } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BAlert, BButton, BFormSelect } from "bootstrap-vue";
+import { computed, onMounted, ref, watch } from "vue";
+
+import { GalaxyApi } from "@/api";
+import { type components } from "@/api/schema";
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import License from "./License.vue";
+import LoadingSpan from "@/components/LoadingSpan.vue";
+
+type LicenseMetadataModel = components["schemas"]["LicenseMetadataModel"];
+
+interface Props {
+    inputLicense: string;
+}
+
+const props = defineProps<Props>();
+
+const emit = defineEmits<{
+    (e: "onLicense", license: string): void;
+}>();
+
+const editLicense = ref(false);
+const licensesLoading = ref(false);
+const errorMessage = ref<string>("");
+const currentLicense = ref<string>(props.inputLicense);
+const licenses = ref<LicenseMetadataModel[] | undefined>([]);
+
+const currentLicenseInfo = computed(() => {
+    for (const l of licenses.value || []) {
+        if (l.licenseId == currentLicense.value) {
+            return l;
+        }
+    }
+
+    return null;
+});
+const licenseOptions = computed(() => {
+    const options = [];
+
+    options.push({
+        value: null,
+        text: "*Do not specify a license.*",
+    });
+
+    for (const l of licenses.value || []) {
+        if (l.licenseId == currentLicense.value || l.recommended) {
+            options.push({
+                value: l.licenseId,
+                text: l.name,
+            });
+        }
+    }
+
+    return options;
+});
+
+function onSave() {
+    onLicense(currentLicense.value);
+    disableEdit();
+}
+
+function disableEdit() {
+    editLicense.value = false;
+    errorMessage.value = "";
+}
+
+function onLicense(l: string) {
+    emit("onLicense", l);
+}
+
+watch(
+    () => props.inputLicense,
+    (newLicense) => {
+        currentLicense.value = newLicense;
+    }
+);
+
+onMounted(async () => {
```

Comment:
Why `onMounted` ? There's no need for that. Use `onMounted` only if you need the DOM before making a request.